manim/renderer/cairo_renderer.py

```diff
@@ -179,30 +179,15 @@ def update_skipping_status(self):
         """
         if config["from_animation_number"]:
             if self.num_plays < config["from_animation_number"]:
-                config["skip_animations"] = True
+                self.skip_animations = True
         if config["upto_animation_number"]:
             if self.num_plays > config["upto_animation_number"]:
-                config["skip_animations"] = True
+                self.skip_animations = True
                 raise EndSceneEarlyException()
 
-    def revert_to_original_skipping_status(self):
-        """
-        Forces the scene to go back to its original skipping status,
-        by setting skip_animations to whatever it reads
-        from original_skipping_status.
-
-        Returns
-        -------
-        Scene
-            The Scene, with the original skipping status.
-        """
-        if hasattr(self, "original_skipping_status"):
-            config["skip_animations"] = self.original_skipping_status
-        return self
-
     def finish(self, scene):
-        config["skip_animations"] = False
+        self.skip_animations = self.original_skipping_status
```

Comment:
**Question** : Are you sure about this? 
If animation n ends and `original_skipping_status` is true, then animation n+1 will have by default `skip_animations` set to true as well, which can be problematic

Reply:
I'm not sure. It makes sense to me because skip_animations is being updated each frame anyway, but it has to keep its original status as a fallback. But I'm not sure. Perhaps @eulertour or @behackl know.

Reply:
The issue is, `original_skipping_status` takes the same value as skip_animation when a new render is started. 
So, if for any reason `original_skipping_status` is set to False it will never change. 

I remember having the exact same problem, and I believ I'm the one who wrote this. 
But maybe with the recent refactor it has changed

Reply:
Sorry for the delay. I found some time today to come back to this. I agree and I believe you. I think this may have some messy behavior. However, I have not been able to find it. I've been messing around with the following scene:

```python
from manim import *

class MyScene(Scene):
    def construct(self):
        t = MathTex('0')
        self.play(ShowCreation(t))
        s = MathTex('1')
        self.play(Uncreate(t), ShowCreation(s))
        t = MathTex('2')
        self.play(Uncreate(s), ShowCreation(t))
        s = MathTex('3')
        self.play(Uncreate(t), ShowCreation(s))
        t = MathTex('4')
        self.play(Uncreate(s), ShowCreation(t))
```

As I render it with the `-n` flag in different combinations, it always seems to work correctly. For example `-n3`, `-n0,3`, `-n3,-1`, `-n2,3` all seem to produce the correct video.

If not with the `-n` flag, can you come up with a way of finding what is the weird behavior that this may cause?

Reply:
Sorry, I was unclear. I meant that when some scenes are rendered in a row, it can cause issues. 

Try with `-a` in a file

Reply:
Oh I see what you mean now. I just tried with a file with two scenes using -a. There is no problem.

Note that now that skip_animations is an instance attribute, there is no way that the skipping status of one scene can affect the skipping status of any other scene that is rendered after (or before).